torch/csrc/jit/serialization/unpickler.cpp

```diff
@@ -868,6 +868,21 @@ void Unpickler::rebuildTensor(bool quantized) {
     impl->set_storage_offset(storage_offset);
     impl->set_sizes_and_strides(size, stride);
     result = autograd::make_variable(result, requires_grad);
+
+    // Handle if math_bits were pickled.
+    // See `args` of _reduce_ex_internal
+    // for a regular tensor (final else case).
+    // Tensors pickled before this patch didn't
+    // have this argument for storing MathBits,
+    // in that case, we do nothing.
+    // NOTE: `math_bits` is the 7th arg.
+    // NOTE: This is only meant for regular tensor and not quantized!
+    if (!quantized && elements.size() == 7) {
```

Comment:
Instead of silently ignoring elements.size(), you should have an assert that it's not quantized when there are too many elements

Reply:
Actually, `quantized` path already has 7 args.
https://github.com/pytorch/pytorch/blob/8f66ae413f8c9d7f2418d7f0b9f69d409c455b46/torch/_tensor.py#L332-L342

Reply:
But... it's a different builder function, no?

Reply:
C++ unpickler shares a common builder function.
https://github.com/pytorch/pytorch/blob/bc66ddb5cb276f3ef0be4d73819f1b172e0872d1/torch/csrc/jit/serialization/unpickler.cpp#L628-L633